clients/src/main/java/org/apache/kafka/common/security/plain/PlainSaslServer.java

```diff
@@ -155,7 +149,7 @@ public SaslServer createSaslServer(String mechanism, String protocol, String ser
             if (!PLAIN_MECHANISM.equals(mechanism)) {
                 throw new SaslException(String.format("Mechanism \'%s\' is not supported. Only PLAIN is supported.", mechanism));
             }
-            return new PlainSaslServer(cbh);
+            return new PlainSaslServer(cbh, new JaasPlainPasswordVerifier());
```

Comment:
@piyushvijay Can you take a look at KIP-86 (https://cwiki.apache.org/confluence/display/KAFKA/KIP-86%3A+Configurable+SASL+callback+handlers) and see if that would work for you?